vortex-array/src/array/varbinview/mod.rs

```diff
@@ -276,20 +270,29 @@ impl VarBinViewArray {
     ///
     /// This is useful for iteration over the values, as well as for applying filters that may
     /// only require hitting the prefixes or inline strings.
-    pub fn view_slice(&self) -> &[BinaryView] {
-        unsafe {
-            slice::from_raw_parts(
-                PrimitiveArray::try_from(self.views())
-                    .vortex_expect("Views must be a primitive array")
-                    .maybe_null_slice::<u8>()
-                    .as_ptr() as _,
-                self.views().len() / VIEW_SIZE_BYTES,
```

Comment:
this is no longer safe to do since the views can be `ConstantArray`. This method was mainly used

1. As an iterator for ArrayAccessor or chunked array packing
2. As a point lookup for `bytes_at`

Instead we split this out into two methods. One that canonicalizes and wraps the Buffer into an iterator, and the other which `slice`s the views buffer to access individual elements